src/raftstore/store/store.rs

```diff
@@ -1468,10 +1469,15 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                 p.heartbeat_pd(&self.pd_worker);
             }
 
+            let peer_id = cp.peer.get_id();
             match change_type {
                 ConfChangeType::AddNode | ConfChangeType::AddLearnerNode => {
-                    // Add this peer to cache and heartbeats.
                     let peer = cp.peer.clone();
+                    if p.peer_id() == peer_id {
+                        p.peer = peer.clone();
```

Comment:
why do we need to update `p.peer`?

Reply:
`peer.is_learner` flag maybe change after apply.